### PR TITLE
asserts,interfaces/policy: allow OR-ing of subrule constraints in plug/slot rules

### PR DESCRIPTION
**Pull request by pedronis**
OR-ing is expressed using lists.

### CODE REVIEW

**Comment by niemeyer** [comment]
Test failure here was due to flakiness being fixed in #2150.

File: asserts/ifacedecls.go [hunk 1]
```diff
@@ -342,19 +343,48 @@ func baseCompileRule(context string, rule interface{}, target rule, subrules []s
...
+		switch x := v.(type) {
+		case nil:
 			v = defaultOutcome[subrule]
 			defaultUsed++
+		case []interface{}:
+			alternatives = true
+			l = x
+		}
+		if l == nil {
+			l = []interface{}{v}
```
Comment (niemeyer) [inline asserts/ifacedecls.go]: Shouldn't this be inside the nil case above?  Is there a situation where we can get `l == nil` as a desired consequence of `x == nil`?

File: asserts/ifacedecls_test.go [hunk 2]
```diff
@@ -541,6 +633,12 @@ func (s *plugSlotRulesSuite) TestCompilePlugRuleErrors(c *C) {
 		{`iface:
   allow-connection: foo`, `allow-connection in plug rule for interface "iface" must be a map or one of the shortcuts 'true' or 'false'`},
 		{`iface:
+  allow-connection:
+    - foo`, `alternative 1 allow-connection in plug rule for interface "iface" must be a map`},
```
Comment (niemeyer) [inline asserts/ifacedecls_test.go]: These might read better as "alternative 1 _of_ allow-connection..."

**Comment by pedronis** [comment]
@jdstrand it would be good if you could give a quick look at this

**Comment by jdstrand** [comment]
Due to sprint constraints (\</rimshot\> :) I did not do a deep code review. However, I did review enough to give a +1 as it matches my understanding for the snap declarations spec.

**Pull request merged**